File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-4.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for tok in tokens:
            if tok == "+":
                res = stack.pop() + stack.pop()
                stack.append(res)
            elif tok == "*":
                res = stack.pop() * stack.pop()
                stack.append(res)
            elif tok == "-":
                val1 = stack.pop()
                val2 = stack.pop()
                stack.append(val2 - val1)
            elif tok == "/":
                val1 = stack.pop()
                val2 = stack.pop()
                stack.append(int(float(val2) / val1))
            else:
                stack.append(int(tok))
        return stack[-1]

File: Data Structures & Algorithms/generate-parentheses/submission-4.js
class Solution {
    /**
     * @param {number} n
     * @return {string[]}
     */
    generateParenthesis(n) {
        const stack = [];
        const res = [];
        dfs(0, 0);
        return res;

        function dfs(openN, closedN) {
            if (closedN === n) {
                res.push(stack.join(""));
                return;
            }

            if (openN < n) {
                stack.push("(");
                dfs(openN + 1, closedN);
                stack.pop();
            }

            if (closedN < openN) {
                stack.push(")");
                dfs(openN, closedN + 1);
                stack.pop();
            }
        }
    }
}
